lib/handlebars/runtime.js

```diff
@@ -74,25 +74,18 @@ export function template(templateSpec, env) {
     }
     partial = env.VM.resolvePartial.call(this, partial, context, options);
 
-    let extendedOptions = Utils.extend({}, options, {
-      hooks: this.hooks,
-      protoAccessControl: this.protoAccessControl
-    });
-
-    let result = env.VM.invokePartial.call(
-      this,
-      partial,
-      context,
-      extendedOptions
-    );
+    options.hooks = this.hooks;
```

Comment:
I guess this proposal will ruin test-coverage, but maybe there is a way around that.

Reply:
Tried profiling this using the script in the linked issue. This PR is 6% faster than using `Object.assign`, and 12% faster than using `Utils.extend`. Note that the options object is already modified in this function in a couple places (`options.ids` and `options.partials`). It's also modified in the `invokePartial` function. My understanding is that `options` is a literal that comes from the template, so it's always new, and this function just augments that. There's no benefit to copying AFAICT.

Reply:
Ok, I can see what you say. I still think that side-effects are bad, but I can understand your motivations.

Its been a long time that I wrote this code, but I think the main reason to use "extend" here was clean code and avoiding side-effects, not some particular bad thing that would happen otherwise.

I wonder what the performance improvement is on a real-life template though.